app/controllers/users_controller.rb

```diff
@@ -2,7 +2,7 @@ class UsersController < ApplicationController
   before_action :authenticate_user!
 
   def index
-    @users = User.all
+    @users = User.includes(:posts)
```

Comment:
- The user model is a special type of model and we conventionally do not fetch other resources along with it as it is used to both authenticate and authorize user identities. You may want to solve N+1 problems with other models that are the core of your logic, in this case, the Post.